vhdl_parser/src/declarative_part.rs

```diff
@@ -176,6 +199,25 @@ package ident is new lib.foo.bar
         );
     }
 
+    #[test]
+    fn parse_declarative_part_recover() {
+        let code = Code::new(
+            "\
+var invalid: broken;
+constant x: natural := 5;
+",
+        );
+        let (decls, msgs) =
+            code.with_partial_stream_messages(parse_declarative_part_leave_end_token);
+        let decls = decls.unwrap();
+        assert_eq!(decls.len(), 1);
+        match decls[0] {
+            Declaration::Object(..) => (),
+            _ => panic!("expected Object declaration, got {:?}", decls[0]),
+        }
```

Comment:
I would prefer if the decls and error messages where tested in full. The `code` utility can be used to create the expected values conveniently.